### PR TITLE
refine agg/topn and add more unit tests

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: ref #5900 

Problem Summary:

### What is changed and how it works?
- move `MergeSortingBlocksBlockInputStream` out of MergeSortingBlockInputStream.cpp
- add unit test for aggregator one/two level merge
- add more unit tests for topn
- Replace all "merge blocks" functions with `Block::mergeBlocks`

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/tests/gtest_topn_executor.cpp [hunk 1]
```diff
@@ -214,5 +245,25 @@ try
...
+    for (const auto & table : tables)
+    {
+        std::vector<size_t> limits{0, 1, 10, 20, 199, 200, 300};
+        for (auto limit_num : limits)
+        {
+            auto request = context
+                               .scan("test_db", table)
+                               .topN("key", false, limit_num)
+                               .build(context);
+            auto expect = executeStreams(request, 1);
```
Comment (xzhangxian1008) [inline dbms/src/Flash/tests/gtest_topn_executor.cpp:261]: more concurrency?

Comment (SeaRise) [reply dbms/src/Flash/tests/gtest_topn_executor.cpp:261]: Various concurrency and block sizes have already been executed in https://github.com/pingcap/tiflash/blob/957d2d41c07d538edab9e47bdbb7eaede753c60b/dbms/src/TestUtils/ExecutorTestUtils.cpp#L105-L146.
Here just to get a baseline of the results.

Comment (xzhangxian1008) [reply dbms/src/Flash/tests/gtest_topn_executor.cpp:261]: > 

okk

**Comment by xzhangxian1008** [comment]
lgtm

**Review by mengxin9014 — APPROVED** [review]
LGTM

**Review by Lloyd-Pottiger — APPROVED** [review]

**Review by LittleFall — APPROVED** [review]

**Pull request merged**